helix-core/src/main/java/org/apache/helix/controller/stages/BestPossibleStateCalcStage.java

```diff
@@ -263,19 +263,6 @@ private boolean computeResourceBestPossibleState(ClusterEvent event, ResourceCon
       } catch (Exception e) {
         LogUtil.logError(logger, _eventId,
             "Error computing assignment for resource " + resourceName + ". Skipping.");
-        // TODO : remove this part after debugging NPE
```

Comment:
Have you add NPE checks for partition assignment?

Reply:
I updated my commit and added NPE checks.